diff --git a/android/src/paper/java/com/facebook/react/viewmanagers/AutoLayoutViewManagerDelegate.java b/android/src/paper/java/com/facebook/react/viewmanagers/AutoLayoutViewManagerDelegate.java
index 4c9080790230cc75f48e4ee68c84df1080986bd6..29f81e03caed5c97401de153c53fe71c3cce9e10 100644
--- a/android/src/paper/java/com/facebook/react/viewmanagers/AutoLayoutViewManagerDelegate.java
+++ b/android/src/paper/java/com/facebook/react/viewmanagers/AutoLayoutViewManagerDelegate.java
@@ -11,10 +11,11 @@ package com.facebook.react.viewmanagers;
 
 import android.view.View;
 import androidx.annotation.Nullable;
+import com.facebook.react.uimanager.BaseViewManager;
 import com.facebook.react.uimanager.BaseViewManagerDelegate;
-import com.facebook.react.uimanager.BaseViewManagerInterface;
+import com.facebook.react.uimanager.LayoutShadowNode;
 
-public class AutoLayoutViewManagerDelegate<T extends View, U extends BaseViewManagerInterface<T> & AutoLayoutViewManagerInterface<T>> extends BaseViewManagerDelegate<T, U> {
+public class AutoLayoutViewManagerDelegate<T extends View, U extends BaseViewManager<T, ? extends LayoutShadowNode> & AutoLayoutViewManagerInterface<T>> extends BaseViewManagerDelegate<T, U> {
   public AutoLayoutViewManagerDelegate(U viewManager) {
     super(viewManager);
   }
diff --git a/android/src/paper/java/com/facebook/react/viewmanagers/CellContainerManagerDelegate.java b/android/src/paper/java/com/facebook/react/viewmanagers/CellContainerManagerDelegate.java
index 2b64af7d5ff936793bf027cba249f31777daba9f..4ba6e5fd815aeac1da56374c8c48a9fcd7f57061 100644
--- a/android/src/paper/java/com/facebook/react/viewmanagers/CellContainerManagerDelegate.java
+++ b/android/src/paper/java/com/facebook/react/viewmanagers/CellContainerManagerDelegate.java
@@ -11,10 +11,11 @@ package com.facebook.react.viewmanagers;
 
 import android.view.View;
 import androidx.annotation.Nullable;
+import com.facebook.react.uimanager.BaseViewManager;
 import com.facebook.react.uimanager.BaseViewManagerDelegate;
-import com.facebook.react.uimanager.BaseViewManagerInterface;
+import com.facebook.react.uimanager.LayoutShadowNode;
 
-public class CellContainerManagerDelegate<T extends View, U extends BaseViewManagerInterface<T> & CellContainerManagerInterface<T>> extends BaseViewManagerDelegate<T, U> {
+public class CellContainerManagerDelegate<T extends View, U extends BaseViewManager<T, ? extends LayoutShadowNode> & CellContainerManagerInterface<T>> extends BaseViewManagerDelegate<T, U> {
   public CellContainerManagerDelegate(U viewManager) {
     super(viewManager);
   }
